gitops/snapshot.go

```diff
@@ -703,3 +706,21 @@ func RemoveIntegrationTestRerunLabel(adapterClient client.Client, ctx context.Co
 
 	return nil
 }
+
+func GetLatestUpdateTime(snapshot *applicationapiv1alpha1.Snapshot) (time.Time, error) {
+	//specific time format choosen, regular format errors out on parsing
+	dateString := "2005/01/02/15:04:05"
+	latestUpdateTime := snapshot.GetAnnotations()[SnapshotPRLastUpdate]
+	if latestUpdateTime == "" {
+		err := metadata.SetAnnotation(snapshot, SnapshotPRLastUpdate, time.Now().String())
```

Comment:
Why Get method should update something?
This is against SRP and it doesn't belong here, setting annotation must be done somewhere else.